### PR TITLE
Minified equippable tests

### PR DESCRIPTION
**Pull request by steven2308**
# Description

This was supposed to include [this commit](https://github.com/rmrk-team/evm/commit/0a46876ad03a568bdc3fe3cda0f2bfbafe976d10) where I add a "minified" equippable core implementation. It simply merges the implementations of MA, Nestable and Equippable and removes intermediate steps and no longer necessary overrides. The code is mostly duplicated but I think it's a fair price at this point that the code is so stable. It does not include any extension or library code since after testing each it proved to not help with the size at all, in some cases it even increased it.

The ready to use implementations for equippable now inherit from this version, which saves 0.184 of size using 200 runs.
This PR includes a new mock and test suit for it, going trough all behaviors for MA, Nestable, Equippable and 721. This tests also run for each lego already but not including them again messes up with coverage and would be dangerous since there could be mistakes when merging the main 3 legos.

# Checklist

- [x] Verified code additions
- [x] Updated NatSpec comments (if applicable)
- [x] Regenerated docs
- [x] Ran prettier
- [x] Added tests to fully cover the changes

### CODE REVIEW

File: test/equippable.ts [hunk 1]
```diff
@@ -132,24 +132,11 @@ async function assetsFixture() {
...
-
-  return { equip, renderUtils };
-}
-
 // --------------- END FIXTURES -----------------------
 
 // --------------- EQUIPPABLE BEHAVIOR -----------------------
 
-describe('EquippableMock with Parts', async () => {
+describe('MinifiedEquippableMock with Parts', async () => {
```
Comment (ThunderDeliverer) [inline test/equippable.ts:139]: I think this shouldn't have been modified, right? Same for other instances in this file.. 🤔

File: test/minifiedEquippable.ts [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+import {
+  RMRKCatalogMock,
+  RMRKMinifiedEquippableMock,
+  RMRKEquipRenderUtils,
+  RMRKMultiAssetRenderUtils,
+} from '../typechain-types';
+// --------------- FIXTURES -----------------------
+
+async function partsFixture() {
+  const catalogSymbol = 'NCB';
```
Comment (ThunderDeliverer) [inline test/minifiedEquippable.ts:34]: Not really relevant, but I suppose the `B` comes from `Base`, right? We could change this to:
```suggestion
  const catalogSymbol = 'NCC';
```

File: test/equippable.ts (hunk 1 above)
Comment (steven2308) [reply test/equippable.ts:139]: Actually yes. I use the name of the mock in the describe to identify it easily when all tests run

File: test/minifiedEquippable.ts [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+  // View
+  const view = <RMRKEquipRenderUtils>await viewFactory.deploy();
+  await view.deployed();
+
+  await setupContextForParts(catalog, neon, neon, mask, mask, mintFromMock, nestMintFromMock);
+  return { catalog, neon, mask, view };
+}
+
+async function slotsFixture() {
+  const catalogSymbol = 'SSB';
```
Comment (ThunderDeliverer) [inline test/minifiedEquippable.ts:68]: This is my final suggestion:

```suggestion
  const catalogSymbol = 'SSC';
```

**Review by ThunderDeliverer — APPROVED** [review]
Just this and then I'm happy! 😁

File: test/equippable.ts (hunk 1 above)
Comment (ThunderDeliverer) [reply test/equippable.ts:139]: Oh, I get it. Ok then 😁

**Pull request merged**